### ISSUE TITLE
unable to sign with gpg

### ISSUE THREAD

**Issue by Fuseteam** [fredldotme/git-confined-snap#5]
when i try to sign my commits i get this: 
![image](https://github.com/user-attachments/assets/2d42da83-1e7f-4beb-8e9e-783ec1de2b7b)

but when i try it witht he snap in devmode i don't

not sure what's up but i suspect we're missing gpg-agent in the snap, ~~not sure how to test it tho~~ ah i need to install without the `--devmode` flag

**Comment by fredldotme** [comment]
Sure the GPG interfaces are connected?

**Comment by Fuseteam** [comment]
yes they are both connected (and no including gpg-agent in the snap didn't help)
![image](https://github.com/user-attachments/assets/84b869ab-3215-436b-b4c2-90c25a6f86b2)

**Comment by fredldotme** [comment]
I'd suggest playing with the env var `GPG_AGENT_INFO` and pointing it to the socket path available within the Snap's file system view.

**Comment by Fuseteam** [comment]
hmmm i'll have to figure out how to do that first haha, is this with or without the agent included?

I guess i start by poking around for socket path

**Comment by fredldotme** [comment]
Without. Also helpful: `snap run --shell git-confined.git`

**Comment by Fuseteam** [comment]
huh poking around in `snap run --shell git-confined` i do not see gpg at all in `/usr/bin` :thinking: 

`snap run --shell git-confined.git` says no app `git` found in git-confined strangely enough xd

**Comment by fredldotme** [comment]
Stop focusing on a host component not being shipped in the snap, the interface between snap-side gpg and host-side gpg-agent is the socket.

**Comment by Fuseteam** [comment]
i'm not looking for gpg-agent fwiw i know it's not in there

**Comment by fredldotme** [comment]
Look in `$SNAP/usr/bin`, don't forget relocation of the binaries.

**Comment by Fuseteam** [comment]
thanks, starting to make more sense now, meanwhile i've found there's more to it then meets they eye: https://github.com/canonical/snapd/pull/7693 access to the socket seems forbidden. 

tho the pull request does mention an successful integration, i'll try to mimick that
